services/wallet/collectibles/manager.go

```diff
@@ -420,50 +406,77 @@ func (o *Manager) FetchCollectibleOwnershipByOwner(ctx context.Context, chainID
 // If asyncFetch is true, empty metadata will be returned for any missing collectibles and an EventCollectiblesDataUpdated will be sent when the data is ready.
 // If asyncFetch is false, it will wait for all collectibles' metadata to be retrieved before returning.
 func (o *Manager) FetchAssetsByCollectibleUniqueID(ctx context.Context, uniqueIDs []thirdparty.CollectibleUniqueID, asyncFetch bool) ([]thirdparty.FullCollectibleData, error) {
-	missingIDs, err := o.collectiblesDataDB.GetIDsNotInDB(uniqueIDs)
+	err := o.FetchMissingAssetsByCollectibleUniqueID(ctx, uniqueIDs, asyncFetch)
 	if err != nil {
 		return nil, err
 	}
 
+	return o.getCacheFullCollectibleData(uniqueIDs)
+}
+
+func (o *Manager) FetchMissingAssetsByCollectibleUniqueID(ctx context.Context, uniqueIDs []thirdparty.CollectibleUniqueID, asyncFetch bool) error {
+	missingIDs, err := o.collectiblesDataDB.GetIDsNotInDB(uniqueIDs)
+	if err != nil {
+		return err
+	}
+
 	missingIDsPerChainID := thirdparty.GroupCollectibleUIDsByChainID(missingIDs)
 
-	group := async.NewGroup(ctx)
-	group.Add(func(ctx context.Context) error {
-		for chainID, idsToFetch := range missingIDsPerChainID {
+	// Atomic group stores the error from the first failed command and stops other commands on error
```

Comment:
hmm if one chain is broken, why not allow the other chains to fetch?

Reply:
hmm true about how it behaved before... Alright, I'll consider later whether we want that behavior to change, but for sure it's better to leave it for a separate PR.

Reply:
it's probably a good idea to keep providers and chains independent from one another